### PR TITLE
Fixed party morale bonus from food variety being too low if variety > 10

### LINKED ISSUE #67: >10 Food Variety Does Not Give Appropriate Morale Bonus
Same as title. When you have oil in your inventory food morale is always +2 regardless of current inventory.

### PR DESCRIPTION
**Pull request by zagganoth**
Fixes #67

### CODE REVIEW

File: src/CommunityPatch/Patches/PartyMoralePatch.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using Helpers;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Map;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Party;
+using TaleWorlds.Library;
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches {
+
+  internal class PartyMoralePatch : PatchBase<PartyMoralePatch> {
```
Comment (tynakuh) [inline src/CommunityPatch/Patches/PartyMoralePatch.cs:18]: A more specific class name might be good idea (Relating to Food Variety), can't imagine it's the only bug around party morale.

File: src/CommunityPatch/Patches/PartyMoralePatch.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo TargetMethodInfo =
+      typeof(DefaultPartyMoraleModel).GetMethod("CalculateFoodVarietyMoraleBonus", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+
+    private static readonly MethodInfo PatchMethodInfo = typeof(StewardAgrarianPatch).GetMethod("Postfix", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
```
[the remark this answers is not in the document]
Comment (TYoung86) [reply src/CommunityPatch/Patches/PartyMoralePatch.cs:25]: Of course if you end up changing the class name to something more specific, adjust as necessary.

[the remark this answers is not in the document]
Comment (Tyler-IN) [reply src/CommunityPatch/Patches/PartyMoralePatch.cs:25]:
```suggestion
    private static readonly MethodInfo PatchMethodInfo = typeof(PartyMoralePatch).GetMethod("Postfix", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
```
Of course if you end up changing the class name to something more specific, adjust as necessary.

[the remark this answers is not in the document]
Comment (Tyler-IN) [reply src/CommunityPatch/Patches/PartyMoralePatch.cs:25]:
```suggestion
    private static readonly MethodInfo PatchMethodInfo = typeof(PartyMoralePatch).GetMethod("Postfix", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
```
Of course if you end up changing the class name to something more specific, adjust as necessary.

File: src/CommunityPatch/Patches/PartyMoralePatch.cs (hunk 1 above)
Comment (Tyler-IN) [reply src/CommunityPatch/Patches/PartyMoralePatch.cs:18]: `MaxFoodVarietyPartyMoralePatch` maybe?

**Comment by Skau** [comment]
> https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/blob/614e7ebc81b2c8af8009f9cce1a4e624ddb56e6f/src/CommunityPatch/Patches/MaxFoodVarietyPartyMoralePatch.cs#L61-L62
> 
> Suggestion: Instead of capping it at 8, maybe continue increasing it from there? Then it would be future-proof if TW added more food.

You can probably just do something like:
```csharp
if (party.ItemRoster.FoodVariety > 10)
    __result = party.ItemRoster.FoodVariety - 3;
```

**Comment by zagganoth** [comment]
> > https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/blob/614e7ebc81b2c8af8009f9cce1a4e624ddb56e6f/src/CommunityPatch/Patches/MaxFoodVarietyPartyMoralePatch.cs#L61-L62
> > 
> > Suggestion: Instead of capping it at 8, maybe continue increasing it from there? Then it would be future-proof if TW added more food.
> 
> You can probably just do something like:
> 
> ```cs
> if (party.ItemRoster.FoodVariety > 10)
>     __result = party.ItemRoster.FoodVariety - 3;
> ```

Done! (Had to do it with result.Add since the patched function doesn't actually have a return value)

**Comment by zagganoth** [comment]
image
Previous food variety morale cap was 8

**Pull request merged**